TestRelease/real_run_absorbs_a_base_branch_that_moved_during_the_build golden is timing-order fragile

## What happens

`erun-integration`'s `TestRelease/real_run_absorbs_a_base_branch_that_moved_during_the_build` fails deterministically on this host, on a clean `origin/main` checkout (`5fe08530`) with no other changes:

```
--- FAIL: TestRelease (10.87s)
    --- FAIL: TestRelease/real_run_absorbs_a_base_branch_that_moved_during_the_build (1.69s)
```

Reproduced 3x via `go test ./... -run 'TestRelease/real_run_absorbs_a_base_branch_that_moved_during_the_build' -count=3` on the branch under work and again on a fresh `git worktree` of `origin/main` — same failure both times, so this is `main`'s own state, not something introduced by any in-flight branch. Found incidentally while running `make check` for #1460 (an unrelated close-confirm.spec.ts fix); not investigated further or fixed there.

## The diff

The golden (`testdata/release/real_run_absorbs_a_base_branch_that_moved_during_the_build.txt`) expects the "step timing" rows under the root in this order:

```
publish [<ELAPSED>]
  ...
post-release-version-bump [<ELAPSED>]
push [<ELAPSED>]
release [<ELAPSED>]
sync-remote [<ELAPSED>]
verify-publication [<ELAPSED>]
```

The actual run instead produces `push` before `post-release-version-bump` — everything else matches.

## Likely cause

`erun-common/timing.go`'s `orderedTimingRows` sorts rows by duration descending, breaking ties by name only when the duration difference is below `timingOrderNoiseFloor` (100ms) — specifically to keep same-run ordering deterministic. This scenario is a **real, non-dry-run** exercise of `erun release` (`erun.Run(t, []string{"release"}, ...)`, no `--dry-run`): the `push` step does a real `git push`, hits a real rejection, and does a real rebase-and-retry (that's the scenario's whole point — a base branch that moved during the build), while `post-release-version-bump` is a single local commit. On a fast host these both land under the 100ms floor and sort alphabetically (`post-release-version-bump` < `push`, matching the golden). On this host, the retry's extra real git round-trip apparently pushes `push`'s measured duration past the 100ms floor relative to `post-release-version-bump`, flipping the sort to duration-order and breaking the byte-exact golden compare.

If that diagnosis is right, this golden is fragile to host speed for a step whose duration is real (not stubbed/instant) — the same "passes on some machines, fails under load/slower hardware" shape as the Playwright specs fixed in #1457/#1459/#1460, just in the Go integration suite instead. Confirm with evidence (e.g. instrument the actual measured durations for `push` and `post-release-version-bump` in a couple of runs) before choosing a fix; possible directions include reworking the retry-and-rebase step to not carry real git latency inside the timed `push` step, or accepting/normalizing rather than order-locking near-floor real-duration siblings in this specific golden.

## Verification

`go test ./... -run 'TestRelease/real_run_absorbs_a_base_branch_that_moved_during_the_build' -count=5` green on this host, ideally checked on more than one host/speed if possible.

--- a/erun-integration/release_test.go
+++ b/erun-integration/release_test.go
@@ -5,6 +5,7 @@ import (
 	"os/exec"
 	"path/filepath"
 	"regexp"
+	"sort"
 	"strings"
 	"testing"
 
@@ -797,8 +798,9 @@ exit 0
 		// git's push-rejection hint block is worded differently across git
 		// versions, so it cannot be part of a reviewed golden; the `! [rejected]`
 		// line and erun's own retry line are what this scenario locks.
+		combined := normalize.Apply(result.Combined, normalize.Replacement{Pattern: regexp.MustCompile(`(?m)^hint:.*\n?`), Token: ""})
 		golden.Equal(t, "release/real_run_absorbs_a_base_branch_that_moved_during_the_build",
-			normalize.Apply(result.Combined, normalize.Replacement{Pattern: regexp.MustCompile(`(?m)^hint:.*\n?`), Token: ""}))
+			canonicalizeReleaseStageTimingOrder(t, combined))
 
 		// Side effects outside the captured streams, one per contract the
 		// absorbed push has to hold: the repository records the release it
@@ -905,6 +907,64 @@ func seedBareOrigin(t *testing.T, setup env.Setup) string {
 	fixture.RunGit(t, setup.Cwd, "remote", "add", "origin", remoteRoot)
 	fixture.RunGit(t, setup.Cwd, "push", "-u", "-q", "origin", "main")
 	return remoteRoot
+}
+
+// releaseStageTimingSiblingsToCanonicalize are this scenario's top-level
+// release-stage timing rows other than "publish", the one stage whose
+// duration (a multi-arch image build plus two chart publishes) always
+// dominates by a wide, host-independent margin. These five run sequentially
+// and each does at most a couple of real git subprocess calls, except
+// "push": this scenario's whole point is a rejected push that rebases and
+// retries, so "push" alone does several more real git subprocess calls than
+// its siblings. erun-common/timing.go's orderedTimingRows sorts by measured
+// duration, tie-breaking by name only when two rows land within a fixed
+// noise floor of each other — so whether "push"'s extra real git latency
+// crosses that floor and reorders it ahead of its siblings depends on how
+// fast git subprocesses run on the host, not on anything this scenario
+// controls. Canonicalizing this block's order before comparing keeps the
+// golden asserting what is actually stable: which five stages ran and
+// roughly how long each took, not which one happened to be a few git calls
+// slower on this particular machine.
+var releaseStageTimingSiblingsToCanonicalize = []string{
+	"post-release-version-bump", "push", "release", "sync-remote", "verify-publication",
+}
+
+// canonicalizeReleaseStageTimingOrder reorders the contiguous
+// releaseStageTimingSiblingsToCanonicalize block within a normalized
+// step-timing table into a fixed, alphabetical order, so the golden compares
+// something that does not depend on real subprocess timing. Called after
+// normalize.Apply, so every row already reads "<name> [<ELAPSED>]". Fails the
+// test loudly, rather than silently comparing the wrong thing, if the block
+// is not exactly the expected five rows in a row — a sign the production
+// output's shape changed and this helper needs to change with it.
+func canonicalizeReleaseStageTimingOrder(t *testing.T, output string) string {
+	t.Helper()
+	wantCount := len(releaseStageTimingSiblingsToCanonicalize)
+	want := make(map[string]bool, wantCount)
+	for _, name := range releaseStageTimingSiblingsToCanonicalize {
+		want["    "+name+" [<ELAPSED>]"] = true
+	}
+	lines := strings.Split(output, "\n")
+	start := -1
+	for i, line := range lines {
+		if want[line] {
+			start = i
+			break
+		}
+	}
+	if start == -1 || start+wantCount > len(lines) {
+		t.Fatalf("release stage timing siblings not found in:\n%s", output)
+	}
+	block := lines[start : start+wantCount]
+	seen := make(map[string]bool, wantCount)
+	for _, line := range block {
+		if !want[line] || seen[line] {
+			t.Fatalf("release stage timing siblings not found as the expected contiguous block in:\n%s", output)
+		}
+		seen[line] = true
+	}
+	sort.Strings(block)
+	return strings.Join(lines, "\n")
 }
 
 // remoteMainSubjects reads origin/main's commit subjects through a second

